homeassistant/components/camera/arlo.py

```diff
@@ -158,15 +161,42 @@ def disable_motion_detection(self):
         self._motion_status = False
         self.set_base_station_mode(ARLO_MODE_DISARMED)
 
+    def set_defined_attr(self, attr_name, attr_value):
+        """Set attribute if attribute is not None (non-subscriptable)."""
+        if attr_value is not None:
+            self.attrs[attr_name] = str(attr_value)
+
     def update(self):
         """Add an attribute-update task to the executor pool."""
-        self.attrs[ATTR_BATTERY_LEVEL] = self._camera.get_battery_level
-        self.attrs[ATTR_BRIGHTNESS] = self._camera.get_battery_level
-        self.attrs[ATTR_FLIPPED] = self._camera.get_flip_state,
-        self.attrs[ATTR_MIRRORED] = self._camera.get_mirror_state,
-        self.attrs[
-            ATTR_MOTION] = self._camera.get_motion_detection_sensitivity,
-        self.attrs[ATTR_POWERSAVE] = POWERSAVE_MODE_MAPPING[
-            self._camera.get_powersave_mode],
-        self.attrs[ATTR_SIGNAL_STRENGTH] = self._camera.get_signal_strength,
-        self.attrs[ATTR_UNSEEN_VIDEOS] = self._camera.unseen_videos
+        base_station = self._camera.base_station
+
+        if not base_station:
+            return None
+
+        base_station.refresh_rate = SCAN_INTERVAL.total_seconds()
+
+        self._camera.update()
+
+        self.set_defined_attr(ATTR_BATTERY_LEVEL,
+                              self._camera.battery_level)
+        self.set_defined_attr(ATTR_BRIGHTNESS,
+                              self._camera.brightness)
+        self.set_defined_attr(ATTR_FLIPPED,
+                              self._camera.flip_state)
+        self.set_defined_attr(ATTR_MIRRORED,
+                              self._camera.mirror_state)
+        self.set_defined_attr(ATTR_MOTION,
+                              self._camera.motion_detection_sensitivity)
+        self.set_defined_attr(ATTR_SIGNAL_STRENGTH,
+                              self._camera.signal_strength)
+        self.set_defined_attr(ATTR_UNSEEN_VIDEOS,
+                              self._camera.unseen_videos)
+
+        powersave_mode = self._camera.powersave_mode
+        if powersave_mode:
+            self.attrs[ATTR_POWERSAVE] = POWERSAVE_MODE_MAPPING[powersave_mode]
+
+        last_refresh = base_station.last_refresh
+        if last_refresh:
+            self.attrs[ATTR_LAST_REFRESH] = datetime.fromtimestamp(
+                last_refresh).strftime("%A, %B %d, %Y %I:%M:%S")
```

Comment:
How often change this value?

Reply:
@pvizeli The updates will run every 90 seconds (interval set at the top of the components).  The way these cameras work, however, all of them will fire an update(), and those usually go to the same base station.  So, if you have 4 Arlo / Arlo Pro cameras, every 90 seconds there will be 4 base station update queries, which is a bit unnecessary.  Do you have any ideas for how to reduce that to 1 call from all camera components?  I don't think the Throttle decorator will work, because each component is technically obeying the 90 second interval - they just aren't aware of one another. My assumption is that the only way to fix this concurrency would be to add a locking mechanism in the PyArlo library itself. @tchellomello Thoughts on that?

Reply:
We should not store timestamps in Home Assistant state attributes, please remove it.

Reply:
@jwillaz  I think your idea is good, however use the timmer as an internal attribute instead of an attribute for HASS state.